### PR TITLE
@craigspaeth => Use DragDrop component for SectionList

### PR DESCRIPTION
**Pull request by eessex**
re: https://github.com/artsy/positron/issues/1144
- Moves drag-drop logic out of section list in favor of DragContainer component
- Adds 'vertical' option to DragContainer, which renders drop targets above or below a dragged-over item depending on user's mouse position

drag-drop

### CODE REVIEW

File: client/apps/edit/components/content/section_list/index.coffee [hunk 1]
```diff
@@ -32,55 +29,37 @@ module.exports = React.createClass
...
+                sections: @props.sections
+                section: section
+                index: i
+                editing: @state.editingIndex is i
+                ref: 'section' + i
+                key: section.cid
+                channel: @props.channel
+                onSetEditing: @onSetEditing
+              }
+              SectionTool { sections: @props.sections, index: i, key: i}
```
Comment (craigspaeth) [inline client/apps/edit/components/content/section_list/index.coffee]: Nitpick: Missing a space at the end of the object literal here `, key: i}`

Comment (eessex) [inline client/apps/edit/components/content/section_list/index.coffee]: Thanks these are updated now

File: client/components/drag_drop/test/index.coffee [hunk 2]
```diff
@@ -79,8 +89,114 @@ describe 'DragDropContainer', ->
...
+      @component.state.should.eql {
+        dragSource: null
+        dragStartY: null
+        dragTarget: null
+        draggingHeight: 0
+        dropPosition: 'top'
+      }
+
+
+describe 'DragDropContainer Vertical', ->
```
Comment (craigspaeth) [inline client/components/drag_drop/test/index.coffee]: Nice set of tests!

File: client/components/drag_drop/drag_target.coffee [hunk 3]
```diff
@@ -1,27 +1,40 @@
...
+      'data-source': @props.activeSource
       style:
-        width: @setWidth()
+        width: @props.width or '100%'
     },
-      if @props.activeTarget and @props.isDraggable
-        div { className: 'drag-placeholder' }
-      @props.children
\ No newline at end of file
+      @renderDropZone() if @props.dropPosition is 'top'
```
Comment (craigspaeth) [inline client/components/drag_drop/drag_target.coffee]: Nice extracting to a method here 👍 w

**Comment by craigspaeth** [comment]
Nice refactor! Very slick UI too. Well done 👏

**Pull request merged**